source/custom/static_features.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package custom
+
+// getStaticFeatures returns statically configured custom features to discover
+// e.g RMDA related features. NFD configuration file may extend these custom features by adding rules.
+func getStaticFeatureConfig() []CustomFeature {
+	return []CustomFeature{
+		CustomFeature{
+			Name: "rdma.capable",
+			MatchOn: MatchRules{
+				PciIdRegex: []string{"15b3:.*"},
+			},
+		},
+		CustomFeature{
+			Name: "rdma.available",
+			MatchOn: MatchRules{
+				LoadedKMod: []string{"ib_core", "rdma_cm"},
```

Comment:
this will need to be updated.
there are two approaches here IMO:
1. define the bare-minimum kernel modules for RDMA applications to be loaded then allow user to add more (e.g rdma_cm, ib_iser)
2.  ensure the same modules as in [rdma-core ](https://github.com/linux-rdma/rdma-core/tree/master/kernel-boot/modules)are loaded which is what majors distros are using

frankly i prefer 2.